components/camel-groovy/src/main/java/org/apache/camel/language/groovy/GroovyShellFactory.java

```diff
@@ -27,4 +29,8 @@ default String getFileName(Exchange exchange) {
         return null;
     }
 
+    default Map<String, Object> getVariables(Exchange exchange) {
```

Comment:
Can you add javadoc with details what is the purpose of this new method

Reply:
Ok, by the way, if still need to push this commit to 3.x ?